Disher/View Controllers/RecipesViewController.m

```diff
@@ -303,48 +313,77 @@ - (void) queryAPIs:(NSString *) input withOption:(NSInteger) option completionHa
+                    recipeName = meal[@"strMeal"];
+                    imageLink = meal[@"strMealThumb"];
+                    mealID = meal[@"idMeal"];
+                    source = @"TheMealDB";
+                    [Recipe getRecipeInfo:mealID withSource:@"TheMealDB" withCompletion:^(NSDictionary * _Nonnull recipeInformation) {
+                        [self.cuisines addObject:recipeInformation[@"strArea"]];
+                        self.temp = @[recipeInformation[@"strArea"]];
+                        NSArray *cuisine = self.temp;
+                        Recipe *newRecipe = [Recipe initWithRecipe:recipeName withURL:imageLink withSource:source withID:mealID withCuisine:cuisine];
+                        [self.tableViewRecipes addObject:newRecipe];
+                        self.unfilteredTableViewRecipes = [self.tableViewRecipes copy];
+                        NSArray *ownedIngredients = [input componentsSeparatedByString:@","];
+                        NSMutableArray *recipeIngredients = [NSMutableArray new];
+                        NSInteger i = 1;
+                        BOOL nullFound = NO;
+                        while (!nullFound) {
+                            NSString *ingredient = recipeInformation[[NSString stringWithFormat:@"strIngredient%ld", i]];
+                            if (ingredient != nil && ![ingredient isEqualToString:@""]) {
+                                [recipeIngredients addObject:ingredient];
+                                i++;
+                            }
+                            else {
+                                nullFound = YES;
+                            }
+                        }
+                        NSUInteger ingredientTotal = recipeIngredients.count;
+                        NSUInteger ownedTotal = 0;
+                        for (NSString *ingredient in ownedIngredients) {
+                            for (NSString *usedIngredient in recipeIngredients) {
+                                if ([usedIngredient rangeOfString:ingredient options:NSCaseInsensitiveSearch].location != NSNotFound) {
+                                    ownedTotal++;
+                                }
+                            }
+                        }
+                        float numeratorFloat = ownedTotal;
+                        float denominatorFloat = ingredientTotal;
+                        NSNumber *percentMatch = [NSNumber numberWithFloat:numeratorFloat / denominatorFloat];
+                        [self.mealDBMatchesValues setObject:percentMatch forKey:mealID];
+                        NSString *matchString = [NSString stringWithFormat:@"You have %lu/%lu ingredients", ownedTotal, ingredientTotal];
+                        [self.mealDBMatches setObject:matchString forKey:mealID];
+                        [self refreshData];
+                    }];
+                }
+            }
+            dispatch_group_leave(group);
+        }];
+    });
```

Comment:
I think this would be a good use case for a few helper functions since the logic here is pretty long. You could break it down both by search type and by logic within the search- something like `handleTitleSearch` and `handleIngredientSearch`, and then within `handleCuisineFilter` and `handleIngredientMatchFiltering` etc. etc. 

This makes the code a lot easier to read